pkg/sync/service.go

```
@@ -232,6 +237,66 @@
+
+		// If the source file is qcow2 when encrypting we need to convert and use its raw image
+		imgInfo, err := util.GetQemuImgInfo(downloadFile)
+		if err != nil {
+			err = fmt.Errorf("failed to detect if file is qcow2 %v", downloadFile)
+			return
+		}
+		if imgInfo.Format == "qcow2" {
+			tmpRawFile = fmt.Sprintf("%v-raw.tmp", downloadFile)
+
+			if err := util.ConvertFromQcow2ToRaw(downloadFile, tmpRawFile); err != nil {
+				err = errors.Wrapf(err, "failed to create raw image from qcow2 image %v", downloadFile)
```

golangci-lint (1.62.2)

ineffectual assignment to err

(ineffassign)